pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix

```diff
@@ -0,0 +1,29 @@
+{ pkgs, nodePackages, makeWrapper, nixosTests, nodejs, stdenv, lib, ... }:
+
+let
+
+  packageName = with lib; concatStrings (map (entry: (concatStrings (mapAttrsToList (key: value: "${key}-${value}") entry))) (importJSON ./package.json));
+
+  ourNodePackages = import ./node-composition.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  };
+in
+ourNodePackages."${packageName}".override {
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ nodePackages.node-gyp-build ];
```

Comment:
I think this is executed at build time so it should go to native.

Reply:
Somebody with NodeJS packaging experience please confirm this.

Reply:
Just do it, run the test, if it succeeds and we are likely fine. :)